clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -778,42 +781,61 @@ void StreamChecker::evalFgetc(const FnDescription *Desc, const CallEvent &Call,
   assertStreamStateOpened(OldSS);
 
```

Comment:
This patch caused a downstream test failure.
Here is the fix:

```suggestion
  // We don't model the buffer, thus they should escape.
  State = Call.invalidateRegions(C.blockCount(), State);
```

Test where it currently breaks:
```c++
void test_double_switch_ok() {
  char buffer[10] = {10};
  FILE *F1 = tmpfile();
  if (!F1)
    return;

  char* s = fgets(buffer, sizeof(buffer), F1); // ok
  if (s) {
    clang_analyzer_dump_char(buffer[0]); // expected-warning {{conj_$}} Should not preserve the previous value, thus should not be 10.
  }
  // Hmm, what do we know of the buffer on failure? IDK.

  fclose(F1);
}
```

Reply:
Will this invalidate only the buffer, not `F1`?

Reply:
At first glance you should be right. However, when I tried it, it didn't break any tests but this one.
I presume there must be something else going on that prevents escaping the stream pointer.
I've not checked.

Alternatively what I did was this:
```c++
[[nodiscard]] static ProgramStateRef
escapeArgsAfterIndex(ProgramStateRef State, CheckerContext &C,
                     const CallEvent &Call, unsigned FirstEscapingArgIndex) {
  const auto *CE = Call.getOriginExpr();
  assert(CE);

  if (Call.getNumArgs() <= FirstEscapingArgIndex)
    return State;

  SmallVector<SVal> EscapingArgs;
  EscapingArgs.reserve(Call.getNumArgs() - FirstEscapingArgIndex);
  for (auto EscArgIdx :
       llvm::seq<int>(FirstEscapingArgIndex, Call.getNumArgs()))
    EscapingArgs.push_back(Call.getArgSVal(EscArgIdx));
  State = State->invalidateRegions(EscapingArgs, CE, C.blockCount(),
                                   C.getLocationContext(),
                                   /*CausesPointerEscape=*/false);
  return State;
}

// at the callsite:
State = escapeArgsAfterIndex(State, C, Call, /*FirstEscapingArgIndex=*/2);
```

Reply:
There should be a same kind of problem with `fread`, or not? (That function was not changed in the recent commits.)

Reply:
The problem should apply to all APIs that potentially write to the passed buffer; including `fread` too.